question2/odbcdemo.c

```diff
@@ -0,0 +1,288 @@
+#include <sqlext.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+
+/*
+** Define Some useful defines
+*/
+#if !defined(NULL)
+#define NULL 0
+#endif
+
+/*
+** function: ODBC_error
+**
+** Purpose: Display to stdout current ODBC Errors
+**
+** Arguments: henv _ ODBC Environment handle.
+** hdbc - ODBC Connection Handle error generated on.
+** hstmt - ODBC SQL Handle error generated on.
+**
+** Returns: void
+**
+*/
+void ODBC_error(                 /* Get and print ODBC error messages */
+                SQLHANDLE henv,  /* ODBC Environment */
+                SQLHANDLE hdbc,  /* ODBC Connection Handle */
+                SQLHANDLE hstmt) /* ODBC SQL Handle */
+{
+    UCHAR sqlstate[10];
```

Comment:
per https://www.ibm.com/docs/en/db2-for-zos/12.0.0?topic=functions-sqlerror-retrieve-error-information, sqlstate only needs to be 5 characters + a null terminator
```suggestion
    UCHAR sqlstate[6];
```

Reply:
this doc is from IBM and not microsoft though, so not 100% sure if it's valid here. the [microsoft doc](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlerror-function?view=sql-server-ver16) only says that this function is deprecated